scripts-and-commands/distributed-deployment/apim-3.2.0/APIM_DEPLOYER_LINUX.sh

```diff
@@ -0,0 +1,455 @@
+  printINFOLog
+}
+
+init(){
+  LOG_STATEMENT="Starting the distributed deployment script of the WSO2 API Manager 3.2.0\n"
+  printINFOLog
+  LOG_STATEMENT="Created by Sumudu Weerasuriya from the Integration CS Team\n"
+  printINFOLog
+
+  for var in "$@"
+    do
+      case "$var" in
+      "-v" | "--version") 
+        printVersion
+        exit
+      ;;
+      "-h" | "--help") 
+        printUsage
+        exit
+      ;;
+    esac
+    done
+
+    if [ -z "$deploymentPattern" ]
+      then
+        LOG_STATEMENT="Deployment pattern is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$extractedPackDirName" ]
+      then
+        LOG_STATEMENT="Extracted ZIP directory name is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$zipFilename" ]
+      then
+        LOG_STATEMENT="Zip file name is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$pathToBasePack" ]
+      then
+        LOG_STATEMENT="Base pack path is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$MYSQL_ROOT_USERNAME" ]
+      then
+        LOG_STATEMENT="MySQL root user name is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$MYSQL_ROOT_PASSWORD" ]
+      then
+        LOG_STATEMENT="MySQL root user password is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$APIM_DB_WSO2AM_320" ]
+      then
+        LOG_STATEMENT="APIM database name is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    if [ -z "$SHARED_DB_WSO2AM_320" ]
+      then
+        LOG_STATEMENT="SHARED database name is not defined\n"
+        printERRLog
+        exit 1
+    fi
+
+    LOG_STATEMENT="\n\nDEPLOYMENT DETAILS \n\n* Deployment Pattern: $deploymentPattern \n* Base Pack Path: $pathToBasePack\n* ZIP Name: $zipFilename\n* Directory Name Of The Extracted Content: $extractedPackDirName \n\n"
+    printConfigDetail
+
+    LOG_STATEMENT="\n\nDATABASE DETAILS \n\n* ROOT User Name: $MYSQL_ROOT_USERNAME\n* APIM Database Name: $APIM_DB_WSO2AM_320\n* Shared Database Name: $SHARED_DB_WSO2AM_320\n\n"
+    printConfigDetail
+
+    case "$deploymentPattern" in
+      $DEPLOYMENT_ACTIVE_ACTIVE) 
+        activeActiveDeployment
+      ;;
+      $DEPLOYMENT_FULL) 
+        fullDistributedDeployment
+      ;;
+      *)
+        LOG_STATEMENT="Invalid Deployment Pattern"
+        printERRLog
+        exit 1;;
+    esac
+}
+
+getSystemTimestamp() {
+	timestamp=`date '+%Y-%m-%d %H:%M:%S' | sed 's/\(:[0-9][0-9][0-9]\)[0-9]*$/\1/' `
+}
+
+printINFOLog(){
+  getSystemTimestamp
+  printf "[${timestamp}] INFO - $LOG_STATEMENT"
+}
+
+printERRLog(){
+  getSystemTimestamp
+  printf "[${timestamp}] ERROR - $LOG_STATEMENT"
+}
+
+printWARNLog(){
+  getSystemTimestamp
+  printf "[${timestamp}] WARN - $LOG_STATEMENT"
+}
+
+printConfigDetail(){
+  getSystemTimestamp
+  printf "[${timestamp}] CONFIG_DETAIL \n$LOG_STATEMENT"
+}
+
+init
```

Comment:
Let's have a new line at the end of each artefact as a practice.

Reply:
This will print log statements regarding all the configuration details as below. (Not similar to the INFO logs)

`
[2022-05-15 18:47:47] CONFIG_DETAIL

NODE DETAILS 

----------- Control Plane -----------

 * Offset: 2
 * Carbon Console URL: https://localhost:9445/carbon
 * Publisher Access URL: https://localhost:9445/publisher
 * Dev Portal Access URL: https://localhost:9445/devportal

----------- Traffic Manager -----------

 * Offset: 1
 * Carbon Console URL: https://localhost:9444/carbon

----------- API Gateway -----------

 * Offset: 0
 * Carbon Console URL: https://localhost:9443/carbon
 * HTTP URL: https://localhost:8280/
 * HTTPS URL: https://localhost:8243/
`